apps/docs/src/content/docs/explainers/async-transform.mdx

```diff
@@ -1,6 +1,281 @@
+	code.
+</Aside>
+
+The gist of it is: you can do a lot of cool things with a Vite plugin and we decided to build the
+Async Transform to make your life (hopefully) easier!
+
+</details>
+
+## How the `Async Transform` changes my code?
+
+The reason why we are writing this explainer is because we firmly believe that if you allow your code
+to be changed by some script you should fully understand how it changes it and what the final result
+will look like.
+
+As shown in our [Mid run cancellation](/explainers/mid-run-cancellation) explainer, this Vite plugin
+aims to solve a couple of problems with async code in Typescript:
+
+-   Promises are non cancellable: once you start them there's no way to stop the code in the middle of
+    the function from executing.
+-   Generators can be "stopped" mid execution but Typescript doesn't play very well with the `yield`
+    keyword.
+
+To fix these problems, we let you write the simple and familiar async code, and we provide a Vite
+plugin to transform your async functions into generators!
+
+<Tabs>
+
+<TabItem label="Before">
+
+```svelte
+<script lang="ts">
+	import { task } from '@sheepdog/svelte';
+
+	const myTask = task(async () => {
+		const result = await fetch('/api/my-endpoint');
+		return await result.json();
+	});
+</script>
+
+<button
+	on:click={() => {
+		myTask.perform();
+	}}
+>
+	perform
+</button>
+```
+
+</TabItem>
+
+<TabItem label="After">
+
+```svelte
+<script lang="ts">
+	import { task } from '@sheepdog/svelte';
+
+	const myTask = task(async function* () {
+		const result = yield fetch('/api/my-endpoint');
+		return yield result.json();
+	});
+</script>
+
+<button
+	on:click={() => {
+		myTask.perform();
+	}}
+>
+	perform
+</button>
+```
+
+</TabItem>
+
+</Tabs>
+
+As you can see we aim to touch your code the bare minimum. But I can still see the worry on your
+face, so let's dive a bit more and let's make a couple of clarifications
+
+### What about my other async functions?
+
+If you don't want to see all of your beloved async functions be turned into generators, worry not,
+because we specifically target only the async functions that are arguments of our own
+`task` or `task.modifier` functions. And we were really careful with this so even if you have other
+functions that are named `task` or if you rename your import everything outside the argument of that
+function will be left untouched.
+
+<Tabs>
+
+<TabItem label="Before">
+
+{/* prettier-ignore-start */}
+
+```svelte {"This will remain untouched": 7-10} {"This will still be transformed": 12-16}
+<script lang="ts">
+	import { task as sheepdogTask } from '@sheepdog/svelte';
+
+	function task(fn: () => Promise<any>) {
+		// your code
+	}
+
+	const otherTask = task(async () => {
+		// some async stuff
+	});
+
+
+	const myTask = sheepdogTask(async () => {
+		const result = await fetch('/api/my-endpoint');
+		return await result.json();
+	});
+</script>
+```
```

Comment:
```suggestion
	```svelte
	<script lang="ts">
		import { task as sheepdogTask } from '@sheepdog/svelte';
	
		// this function will remain untouched
		function task(fn: () => Promise<any>) {
			// your code
		}
	
		// this will be transformed
		const otherTask = task(async () => {
			// some async stuff
		});
	
	
		const myTask = sheepdogTask(async () => {
			const result = await fetch('/api/my-endpoint');
			return await result.json();
		});
	</script>
	```
```


Another nitpick but I found this to be slightly clearer